### ISSUE TITLE
Error building FFmpeg extension

### ISSUE THREAD

**Issue by AOrobator** [google/ExoPlayer#2769]
I get the following error when I attempt to build the FFmpeg extension for ExoPlayer 2.4.0

`Unknown option "--target-os=android"`

I'm on a Mac running Sierra and have NDK r14b

I'm using the following to build/make the FFmpeg extension:

```
COMMON_OPTIONS="\
    --target-os=android \
    --disable-static \
    --enable-shared \
    --disable-doc \
    --disable-programs \
    --disable-everything \
    --disable-avdevice \
    --disable-avformat \
    --disable-swscale \
    --disable-postproc \
    --disable-avfilter \
    --disable-symver \
    --disable-swresample \
    --enable-avresample \
    --enable-decoder=vorbis \
    --enable-decoder=opus \
    --enable-decoder=flac \
    -–enable-decoder=alac \
    -–enable-decoder=mp3 \
    -–enable-decoder=aac \
    " && \
cd "${FFMPEG_EXT_PATH}/jni" && \
git clone git://source.ffmpeg.org/ffmpeg ffmpeg && cd ffmpeg && \
./configure \
    --libdir=android-libs/armeabi-v7a \
    --arch=arm \
    --cpu=armv7-a \
    --cross-prefix="${NDK_PATH}/toolchains/arm-linux-androideabi-4.9/prebuilt/${HOST_PLATFORM}/bin/arm-linux-androideabi-" \
    --sysroot="${NDK_PATH}/platforms/android-9/arch-arm/" \
    --extra-cflags="-march=armv7-a -mfloat-abi=softfp" \
    --extra-ldflags="-Wl,--fix-cortex-a8" \
    --extra-ldexeflags=-pie \
    ${COMMON_OPTIONS} \
    && \
make -j4 && make install-libs && \
make clean && ./configure \
    --libdir=android-libs/arm64-v8a \
    --arch=aarch64 \
    --cpu=armv8-a \
    --cross-prefix="${NDK_PATH}/toolchains/aarch64-linux-android-4.9/prebuilt/${HOST_PLATFORM}/bin/aarch64-linux-android-" \
    --sysroot="${NDK_PATH}/platforms/android-21/arch-arm64/" \
    --extra-ldexeflags=-pie \
    ${COMMON_OPTIONS} \
    && \
make -j4 && make install-libs && \
make clean && ./configure \
    --libdir=android-libs/x86 \
    --arch=x86 \
    --cpu=i686 \
    --cross-prefix="${NDK_PATH}/toolchains/x86-4.9/prebuilt/${HOST_PLATFORM}/bin/i686-linux-android-" \
    --sysroot="${NDK_PATH}/platforms/android-9/arch-x86/" \
    --extra-ldexeflags=-pie \
    --disable-asm \
    ${COMMON_OPTIONS} \
    && \
make -j4 && make install-libs && \
make clean
```

**Comment by andrewlewis** [comment]
I just went through the instructions in the [README](https://github.com/google/ExoPlayer/tree/release-v2/extensions/ffmpeg) on a Mac and it worked fine. I suggest starting from the instructions in there and making the modifications you need.

This might be unrelated, but I did notice that your pasted configuration includes what look like en-dashes on the lines for enabling the MP3 and AAC decoders. You probably want to replace those with hyphens.

**Comment by AOrobator** [comment]
I copied and pasted the exact commands from the readme and I get the same error. I'm pasting them here as well for completion. Is it possible I don't have something installed or my path variables are incorrect?

### Command Used

```
COMMON_OPTIONS="\
    --target-os=android \
    --disable-static \
    --enable-shared \
    --disable-doc \
    --disable-programs \
    --disable-everything \
    --disable-avdevice \
    --disable-avformat \
    --disable-swscale \
    --disable-postproc \
    --disable-avfilter \
    --disable-symver \
    --disable-swresample \
    --enable-avresample \
    --enable-decoder=vorbis \
    --enable-decoder=opus \
    --enable-decoder=flac \
    " && \
cd "${FFMPEG_EXT_PATH}/jni" && \
git clone git://source.ffmpeg.org/ffmpeg ffmpeg && cd ffmpeg && \
./configure \
    --libdir=android-libs/armeabi-v7a \
    --arch=arm \
    --cpu=armv7-a \
    --cross-prefix="${NDK_PATH}/toolchains/arm-linux-androideabi-4.9/prebuilt/${HOST_PLATFORM}/bin/arm-linux-androideabi-" \
    --sysroot="${NDK_PATH}/platforms/android-9/arch-arm/" \
    --extra-cflags="-march=armv7-a -mfloat-abi=softfp" \
    --extra-ldflags="-Wl,--fix-cortex-a8" \
    --extra-ldexeflags=-pie \
    ${COMMON_OPTIONS} \
    && \
make -j4 && make install-libs && \
make clean && ./configure \
    --libdir=android-libs/arm64-v8a \
    --arch=aarch64 \
    --cpu=armv8-a \
    --cross-prefix="${NDK_PATH}/toolchains/aarch64-linux-android-4.9/prebuilt/${HOST_PLATFORM}/bin/aarch64-linux-android-" \
    --sysroot="${NDK_PATH}/platforms/android-21/arch-arm64/" \
    --extra-ldexeflags=-pie \
    ${COMMON_OPTIONS} \
    && \
make -j4 && make install-libs && \
make clean && ./configure \
    --libdir=android-libs/x86 \
    --arch=x86 \
    --cpu=i686 \
    --cross-prefix="${NDK_PATH}/toolchains/x86-4.9/prebuilt/${HOST_PLATFORM}/bin/i686-linux-android-" \
    --sysroot="${NDK_PATH}/platforms/android-9/arch-x86/" \
    --extra-ldexeflags=-pie \
    --disable-asm \
    ${COMMON_OPTIONS} \
    && \
make -j4 && make install-libs && \
make clean
```

### .zshrc

```
# Path to your oh-my-zsh installation.
export ZSH=/Users/AndrewOrobator/.oh-my-zsh

# Set name of the theme to load.
# Look in ~/.oh-my-zsh/themes/
# Optionally, if you set this to "random", it'll load a random theme each
# time that oh-my-zsh is loaded.
ZSH_THEME="agnoster"

plugins=(git osx zsh-autosuggestions)

# User configuration

export PATH="/usr/bin:/bin:/usr/sbin:/sbin:$PATH"
# export MANPATH="/usr/local/man:$MANPATH"

source $ZSH/oh-my-zsh.sh

alias c="clear"
alias d="adb devices"
alias gs="git status"
alias gas="git add *"
alias gu="git up"
alias grh="git reset HEAD --hard"
alias gch="git checkout"
alias log="adb logcat"
alias adbr="adb kill-server; adb start-server; d"
alias abdr="adbr"
alias run="./gradlew :app:installDevDebug && adb shell am start -n com.orobator.auracle.debug/com.orobator.auracle.collections.MusicActivity"
alias runs="./gradlew :app:installDevDebug -s && adb shell am start -n com.orobator.auracle.debug/com.orobator.auracle.collections.MusicActivity"
alias zshrc="vim ~/.zshrc; sauce"
alias sauce="source ~/.zshrc"
eval "$(thefuck --alias)"

export ANDROID_HOME=/Users/AndrewOrobator/Development/Android/android-sdk-macosx
export ANDROID_SDK=$ANDROID_HOME
export ANDROID_TOOLS=$ANDROID_HOME/tools
export ANDROID_PLATFORM_TOOLS=$ANDROID_HOME/platform-tools
export ANDROID_BUILD_TOOLS_VERSION=`ls $ANDROID_HOME/build-tools | sort -r | head -n 1`
export ANDROID_BUILD_TOOLS_HOME=${ANDROID_HOME}/build_tools/${ANDROID_BUILD_TOOLS_VERSION}

export PATH=$PATH:$ANDROID_TOOLS
export PATH=$PATH:$ANDROID_PLATFORM_TOOLS
launchctl setenv ANDROID_HOME $ANDROID_HOME

export EXOPLAYER_ROOT="/Users/AndrewOrobator/Development/Projects/ExoPlayer"
export FFMPEG_EXT_PATH="${EXOPLAYER_ROOT}/extensions/ffmpeg/src/main"
export NDK_PATH="$ANDROID_HOME/ndk-bundle"
export HOST_PLATFORM="darwin-x86_64"


test -e "${HOME}/.iterm2_shell_integration.zsh" && source "${HOME}/.iterm2_shell_integration.zsh"

```

**Comment by andrewlewis** [comment]
Please run `head -n 1 ${FFMPEG_EXT_PATH}/jni/ffmpeg/ffbuild/config.log` and paste the output here. This should show the arguments passed to FFmpeg's configure script.

**Comment by AOrobator** [comment]
Doesn't look like that file exists. When I run the `head` command, I get the following:

`head: /Users/AndrewOrobator/Development/Projects/ExoPlayer/extensions/ffmpeg/src/main/jni/ffmpeg/ffbuild/config.log: No such file or directory`

The ffbuild directory exists though.
<img width="414" alt="screen shot 2017-05-05 at 22 39 29" src="https://cloud.githubusercontent.com/assets/4106884/25763528/d6c9ba06-31e3-11e7-83e2-a6195042530f.png">

**Comment by AOrobator** [comment]
Just did a bit more debugging.

I deleted the first line of COMMON_OPTIONS, `--target-os=android \` and I got the error 

`Unknown option "--disable-static"`

`--disable-static` is the very next command, so the error might have to do with how the command is formatted.

I also ran `./configure` by itself and it complained that yasm wasn't installed so after doing `brew install yasm`, `./configure` ran to completion.
